### PR TITLE
home-assistant: 2021.3.4 -> 2021.4.0

### PR DESCRIPTION
**Pull request by mweinelt**
###### Motivation for this change

Waiting for #118176 @fabaff 

https://www.home-assistant.io/blog/2021/04/07/release-20214/

nixosTests.home-assistant probably still broken because of #110863. No idea what is wrong there.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

```
       Outdated dependencies        
┏━━━━━━━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━┓
┃ Package       ┃ Current ┃ Wanted ┃
┡━━━━━━━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━┩
│ numpy         │ 1.20.1  │ 1.20.2 │
│ pillow        │ 8.1.1   │ 8.1.2  │
│ plexapi       │ 4.5.0   │ 4.5.1  │
│ plexwebsocket │ 0.0.12  │ 0.0.13 │
└───────────────┴─────────┴────────┘
```

### CODE REVIEW

**Comment by mweinelt** [comment]
Marked as draft, given that we wait on a dependency and that I want to enable a bunch more component tests later tonight.

**Comment by fabaff** [comment]
New dependencies:

- aiodiscover: https://github.com/NixOS/nixpkgs/pull/118176
- yalexs: https://github.com/NixOS/nixpkgs/pull/118766
- screenlogicpy: https://github.com/NixOS/nixpkgs/pull/118767

Updates:

- plexapi, plexwebsocket: #118436

**Comment by mweinelt** [comment]
WIP.

```
=========================== short test summary info ============================
FAILED tests/util/test_package.py::test_check_package_version_does_not_match
FAILED tests/components/axis/test_binary_sensor.py::test_binary_sensors[pyloop]
FAILED tests/components/axis/test_camera.py::test_camera[pyloop] - AssertionE...
FAILED tests/components/axis/test_camera.py::test_camera_with_stream_profile[pyloop]
FAILED tests/components/axis/test_device.py::test_device_setup[pyloop] - KeyE...
FAILED tests/components/axis/test_device.py::test_device_info[pyloop] - KeyEr...
FAILED tests/components/axis/test_init.py::test_setup_entry[pyloop] - KeyErro...
FAILED tests/components/axis/test_device.py::test_device_support_mqtt[pyloop]
FAILED tests/components/axis/test_light.py::test_no_light_entity_without_light_control_representation[pyloop]
FAILED tests/components/axis/test_init.py::test_setup_entry_fails[pyloop] - K...
FAILED tests/components/axis/test_device.py::test_update_address[pyloop] - Ke...
FAILED tests/components/axis/test_light.py::test_lights[pyloop] - KeyError: '...
FAILED tests/components/axis/test_init.py::test_unload_entry[pyloop] - KeyErr...
FAILED tests/components/axis/test_device.py::test_device_unavailable[pyloop]
FAILED tests/components/axis/test_init.py::test_migrate_entry[pyloop] - Asser...
FAILED tests/components/axis/test_device.py::test_device_reset[pyloop] - KeyE...
FAILED tests/components/axis/test_switch.py::test_switches_with_port_cgi[pyloop]
FAILED tests/components/axis/test_device.py::test_device_not_accessible[pyloop]
FAILED tests/components/axis/test_switch.py::test_switches_with_port_management[pyloop]
FAILED tests/components/axis/test_device.py::test_device_trigger_reauth_flow[pyloop]
FAILED tests/components/axis/test_device.py::test_device_unknown_error[pyloop]
FAILED tests/components/default_config/test_init.py::test_setup[pyloop] - ass...
ERROR tests/components/axis/test_config_flow.py
ERROR tests/components/dhcp/test_init.py
ERROR tests/components/verisure/test_config_flow.py
```

`axis` errors are all kind of the same

```
>       assert not hass.data[AXIS_DOMAIN]
E       KeyError: 'axis'
```

**Comment by mweinelt** [comment]
```
=========================== short test summary info ============================
FAILED tests/util/test_package.py::test_check_package_version_does_not_match
FAILED tests/components/litterrobot/test_sensor.py::test_waste_drawer_sensor[pyloop]
FAILED tests/components/litterrobot/test_sensor.py::test_sleep_time_sensor_with_none_state[pyloop]
FAILED tests/components/screenlogic/test_config_flow.py::test_form_cannot_connect[pyloop]
ERROR tests/components/litterrobot/test_vacuum.py::test_vacuum_with_error[pyloop]
= 4 failed, 7763 passed, 5 skipped, 3371 warnings, 1 error in 495.78s (0:08:15) =
```

**Comment by mweinelt** [comment]
Down to one failing test

```
__________________ test_check_package_version_does_not_match ___________________
[gw30] linux -- Python 3.8.8 /nix/store/5470xw15wnn972ap0c4f7q642z4nvh6f-python3-3.8.8/bin/python3.8

    def test_check_package_version_does_not_match():
        """Test for version mismatch."""
        installed_package = list(pkg_resources.working_set)[0].project_name
        assert not package.is_installed(f"{installed_package}==999.999.999")
>       assert not package.is_installed(f"{installed_package}>=999.999.999")
E       AssertionError: assert not True
E        +  where True = <function is_installed at 0x7ffff45e4040>('homeassistant>=999.999.999')
E        +    where <function is_installed at 0x7ffff45e4040> = package.is_installed

tests/util/test_package.py:257: AssertionError
```

**Comment by mweinelt** [comment]
I feel the acl pattern in our home-assistant test doesn't work anymore. But that's just an educated guess.

File: pkgs/development/python-modules/aiodiscover/default.nix [hunk 1]
```diff
@@ -10,24 +10,16 @@
...
   src = fetchFromGitHub {
     owner = "bdraco";
     repo = pname;
     rev = "v${version}";
-    sha256 = "0qg2wm6ddsfai788chylr5ynrvakwg91q3dszz7dxzbkfdcxixj3";
+    sha256 = "186agrjx818vn00d3pqlka5ir48rgpbfyn1cifkn9ylsxg9cz3ph";
   };
 
-  patches = [
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/aiodiscover/default.nix:24]: Please remove fetchpatch from inputs.

**Pull request merged**